compress.cc

```diff
@@ -24,6 +24,7 @@
 #include "exceptions/exceptions.hh"
 #include "utils/class_registrator.hh"
 #include "gms/feature_service.hh"
+#include "db/config.hh"
```

Comment:
It's better to avoid dependencies on db/config.hh and instead have each module export its own local config struct, and have main/cql_test_env stitch them together.

Reply:
This mode does have a local config. The only dependency here is `default_sstable_compressor_factory_config::from_db_config` which extracts the local config from the global config, just so that's not duplicated in both `main` and `cql_test_env`.

Should I inline it into `main` and `cql_test_env`, then?

Reply:
Okay, I moved the definition of the glue method to `init.cc`. (I left the declaration in the same place. I can see some merit in avoiding a header dependency, but I see no merit in avoiding a forward decl.)